examples/trading/api/README.md

```diff
@@ -0,0 +1,120 @@
+# Escrow API Demo
+
+This demo is built to showcase how we can build an event indexer + API
+to easily & efficiently serve on-chain data for our app.
+
+The demo indexer uses polling to watch for new events.
+
+Everything is pre-configured on testnet, but can be easily tweaked to work on any other network.
+
+## Installation
+
+1. Install dependencies by running
+```
+pnpm install
+```
+2. Build BCS & TS SDK in the repo:
+```
+pnpm -w bcs build && pnpm -w sdk build
+```
+3. Setup the database by running
+```
+pnpm db:setup:dev
+```
+4. [Publish the contract & demo data](#demo-data)
+
+5. Run both the API and the indexer
+```
+pnpm dev
+```
+
+6. Visit [http://localhost:3000/escrows](http://localhost:3000/escrows) or [http://localhost:3000/locked](http://localhost:3000/locked)
+
+## Demo Data<a name="demo-data"></a>
+
+> Make sure you have enough testnet (or any net) SUI in the active address of the CLI.
+
+There are some helper functions to:
+1. Publish the smart contract
+2. Create some demo data (for testnet)
+
+To produce demo data you should:
+
+1. Publish the smart contract by running
+
+```
+npx ts-node helpers/publish-contracts.ts
+```
+
+2. Produce demo non-locked and locked objects
+
+```
+npx ts-node helpers/create-demo-data.ts
+```
+
+3. Produce demo escrows
+```
+npx ts-node helpers/create-demo-escrows.ts
+```
+
+
+
+If you want to reset the database (start from scratch), run:
+```
+pnpm db:reset:dev && pnpm db:setup:dev
+```
+
+## API
+
+The API exposes data written from the event indexer.
+
+For each request, we have pagination with a max limit of 50 per page.
+
+| Parameter 	| Expected value  	|
+|-----------	|-----------------	|
+| limit     	| number (1-50)     |
+| cursor    	| number          	|
+| sort      	| 'asc' \| 'desc' 	|
+
+
+There are two available routes:
+
+### `/locked`: Returns indexed locked objects
+
+Available query parameters:
+
+| Parameter 	| Expected value    	|
+|-----------	|-------------------	|
+| deleted   	| 'true' \| 'false' 	|
+| keyId     	| string            	|
+| creator   	| string            	|
+
+### `/escrows`: Returns indexed escrow objects
+
+Available query parameters:
+
+| Parameter 	| Expected value    	|
+|-----------	|-------------------	|
+| cancelled 	| 'true' \| 'false' 	|
+| swapped   	| 'true' \| 'false' 	|
+| recipient 	| string            	|
+| sender    	| string            	|
+
+Example query: 
+
+`GET /escrows?sort=asc&limit=10&cursor=15&recipient=0x2&cancelled=false`
```

Comment:
Would it be possible to format this as a `curl` request for ease of copying? Also could you accompany this with a short description of what it does?

Reply:
Absolutely, will do!